Nfield.Quota/QuotaFrameValidator.cs

```diff
@@ -431,6 +434,54 @@ public QuotaFrameValidator()
             return isValid;
         }
 
+        private static bool HaveNestedMinLevelsSumToLessThanMaxTargetForEachLevel(
+            QuotaFrame frame,
+            IEnumerable<QuotaFrameVariable> variables,
+            PropertyValidatorContext context)
+        {
+            bool ProcessLevel(QuotaFrameVariable variable, IEnumerable<QuotaFrameLevel> parents)
+            {
+                // sum of targets, ignoring null values
+                var sum = variable.Levels.Aggregate(0, (total, level) => total + (level.Target ?? 0));
+
+                var isValid = true;
+                foreach (var parent in parents)
+                {
+                    if (parent.MaxTarget < sum)
+                    {
+                        context.MessageFormatter.AppendArgument("VariableName", variable.Name);
+                        context.MessageFormatter.AppendArgument("VariableId", variable.Id);
+                        context.MessageFormatter.AppendArgument("LevelId", parent.Id);
+                        context.MessageFormatter.AppendArgument("LevelName", parent.Name);
+                        context.MessageFormatter.AppendArgument("Sum", sum);
+                        context.MessageFormatter.AppendArgument("MaxTarget", parent.MaxTarget);
+
+                        isValid = false;
+                    }
+                }
+
+                foreach (var level in variable.Levels)
+                {
+                    var newParents = parents.Concat(new[] { level });
```

Comment:
you concat the parents, so everytime all parents are there. Wondering if you don't get the same message over and over again if the tree is deep, and the invalid parent is high in the tree?

Reply:
This doesn't happen because only the last message is kept. Even so, the message would be different every time because we'd be evaluating a different variable, so the sum would be different. So technically they are different errors (even if all related to the same levels).